src/components/CallView/CallView.vue

```diff
@@ -1,75 +1,310 @@
+import LocalCallParticipantModel from '../../utils/webrtc/models/LocalCallParticipantModel'
+import CallParticipantCollection from '../../utils/webrtc/models/CallParticipantCollection'
 
 export default {
+
 	name: 'CallView',
 
 	components: {
-		Avatar,
-		MediaControls,
+		LocalVideo,
+		Screen,
+		Video,
+	},
+
+	props: {
+		token: {
+			type: String,
+			required: true,
+		},
+		signaling: {
+			type: Object,
+			required: true,
+		},
+	},
+
+	data() {
+		return {
+			speakers: [],
+			speakingUnwatchers: {},
+			screens: [],
+			screenUnwatchers: {},
+			// callParticipantModelsWithScreen: [],
+			localSharedData: {
+				screenVisible: true,
+			},
+			sharedDatas: {},
+
+			localMediaModel: {},
+			localCallParticipantModel: {},
+			callParticipantCollection: {
+				callParticipantModels: [],
+			},
+		}
 	},
 
 	computed: {
-		actorType() {
-			return this.$store.getters.getActorType()
+
+		callViewClass() {
+			const callViewClass = {
+				'incall': this.remoteParticipantsCount > 0,
+				'screensharing': this.screenSharingActive,
+			}
+			callViewClass['participants-' + (this.remoteParticipantsCount + 1)] = true
+
+			return callViewClass
 		},
-		actorId() {
-			return this.$store.getters.getActorId()
+
+		reversedCallParticipantModels() {
+			return this.callParticipantCollection.callParticipantModels.slice().reverse()
 		},
-		displayName() {
-			return this.$store.getters.getDisplayName()
+
+		remoteParticipantsCount() {
+			return this.callParticipantCollection.callParticipantModels.length
+		},
+
+		callParticipantModelsWithScreen() {
+			return this.callParticipantCollection.callParticipantModels.filter(callParticipantModel => callParticipantModel.attributes.screen)
+		},
+
+		screenSharingActive() {
+			return this.screens.length > 0
 		},
 
-		firstLetterOfGuestName() {
-			const customName = this.displayName !== t('spreed', 'Guest') ? this.displayName : '?'
-			return customName.charAt(0)
+		darkBackground() {
+			return this.remoteParticipantsCount > 0 || this.screenSharingActive
 		},
+
 	},
-}
-</script>
 
-<style lang="scss" scoped>
-.stage {
-	background: #000000;
-	width: 100%;
-	height: 100%;
+	watch: {
+
+		'darkBackground': function(darkBackground) {
+			this.$emit('hasDarkBackground', darkBackground)
+		},
+
+		'localMediaModel.attributes.localScreen': function(localScreen) {
+			this._setScreenAvailable(this.localCallParticipantModel.attributes.peerId, localScreen)
+		},
 
-	.promoted-stream {
-		height: auto;
-	}
+		'callParticipantCollection.callParticipantModels': function(models) {
+			const addedModels = models.filter(model => !this.sharedDatas[model.attributes.peerId])
+			const removedModelIds = Object.keys(this.sharedDatas).filter(sharedDataId => models.find(model => model.attributes.peerId === sharedDataId) === undefined)
 
-	.video-container-row {
-		height: 180px;
-		position: absolute;
-		bottom: 0;
-		width: 100%;
+			removedModelIds.forEach(removedModelId => {
```

Comment:
Avoid putting big functions into watchers, put them into methods out of readibility ;)